test/unit/specs/lcdClientMock.spec.js

```diff
@@ -884,6 +884,269 @@ describe(`LCD Client Mock`, () => {
+            proposal_id: `5`,
+            amount: [
+              {
+                denom: `pump&DumpCoin`,
+                amount: `10`
+              }
+            ],
+            depositer: lcdClientMock.addresses[0]
+          })
+          expect(res).toHaveLength(1)
+          expect(res[0].check_tx.code).toBe(1)
+          expect(res[0].check_tx.log).toBe(
+            `Not enough pump&DumpCoins in your account`
+          )
+
+          // user has not enough coins
+          res = await client.submitProposalDeposit({
+            base_req: {
+              name: `default`,
+              sequence: 1
+            },
+            proposal_id: `5`,
+            amount: [
+              {
+                denom: `steak`,
+                amount: `1500`
+              }
+            ],
+            depositer: lcdClientMock.addresses[0]
+          })
+          expect(res).toHaveLength(1)
+          expect(res[0].check_tx.code).toBe(1)
+          expect(res[0].check_tx.log).toBe(`Not enough steaks in your account`)
+        })
+      })
+
+      describe(`submits successfully a deposit on an active proposal`, () => {
+        it(`when user hasn't submitted a previous deposit`, async () => {
+          let proposalBefore = await client.getProposal(`5`)
+          let totalDepositBefore = proposalBefore.total_deposit.find(coin => {
+            return coin.denom === `steak`
+          })
+
+          let userDepositBefore = await client.getProposalDeposit(
+            `5`,
+            lcdClientMock.addresses[0]
+          )
+          expect(totalDepositBefore).not.toBeDefined()
+          expect(userDepositBefore).not.toBeDefined()
+
+          let deposit = {
+            proposal_id: `5`,
+            amount: [
+              {
+                denom: `steak`,
+                amount: `200`
+              }
+            ],
+            depositer: lcdClientMock.addresses[0]
+          }
+
+          let res = await client.submitProposalDeposit({
+            base_req: {
+              name: `default`,
+              sequence: 1
+            },
+            ...deposit
+          })
+          expect(res).toBeDefined()
+
+          // sshould have added the deposit to the proposal's total deposit
+          let proposalAfter = await client.getProposal(`5`)
+          let totalDepositAfter = proposalAfter.total_deposit.find(coin => {
+            return coin.denom === `steak`
+          })
+          expect(totalDepositAfter).toEqual(deposit.amount[0])
+
+          // should have updated the status of the proposal from `Pending` to `Active`
+          expect(proposalAfter.proposal_status).toEqual(`Active`)
+
+          // should have added the deposit from the depositer
+          let userDepositAfter = await client.getProposalDeposit(
+            `5`,
+            lcdClientMock.addresses[0]
+          )
+          expect(userDepositAfter).toEqual(deposit)
+        })
+
+        it(`when user has previous deposit`, async () => {
```

Comment:
I really like the very understandable test description! Great work.

Reply:
haha otherwise I couldn't understand it either